src/librbd/internal.cc

```diff
@@ -2797,6 +2822,58 @@ int validate_mirroring_enabled(ImageCtx *ictx) {
       }
     }
 
+    if (next_mirror_mode == cls::rbd::MIRROR_MODE_POOL) {
+      r = cls_client::mirror_mode_set(&io_ctx, cls::rbd::MIRROR_MODE_IMAGE);
+      if (r < 0) {
+        lderr(cct) << "Failed to set mirror mode to image: "
+          << cpp_strerror(r) << dendl;
+        return r;
+      }
+
+      map<string, string> images;
+      r = list_images_v2(io_ctx, &images);
+      if (r < 0) {
+        lderr(cct) << "Failed listing images: " << cpp_strerror(r) << dendl;
+        r = cls_client::mirror_mode_set(&io_ctx, current_mirror_mode);
+        if (r < 0) {
+          lderr(cct) << "Failed to rollback mirror mode: " << cpp_strerror(r)
+            << dendl;
+        }
+        return r;
+      }
+
+      for (const auto& img_pair : images) {
+        ImageCtx *img_ctx = new ImageCtx("", img_pair.second, nullptr, io_ctx,
+                                         false);
+        r = img_ctx->state->open();
+        if (r < 0) {
+          lderr(cct) << "error opening image "<< img_pair.first << ": "
+            << cpp_strerror(r) << dendl;
+          delete img_ctx;
+          continue;
+        }
+
+        uint64_t features;
+        r = get_features(img_ctx, &features);
```

Comment:
Slight simplification: you have inside librbd and have direct access to the ImageCtx.  Just use `ImageCtx::test_features`